core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -214,8 +230,14 @@ class KryoSerializer(conf: SparkConf)
     kryo
   }
 
+  override def setDefaultClassLoader(classLoader: ClassLoader): Serializer = {
+    defaultClassLoader = Some(classLoader)
```

Comment:
You can write `super.setDefaultClassLoader(classLoader)` here to inherit the behavior rather than duplicate it. It is just one line now, yes.

Where is defaultClassLoader used in this implementation though? I wonder why it matters that you call `getPool` after this field is set. And if it does, isn't setting it in line 105 not actually helping?

Reply:
defaultClassLoader is used in newKryo.

I called `getPool` after setting the defaultClassLoader to make sure we don't accidently create a newKryo before the defaultClassLoader is updated. Setting it on 105 was because I don't believe `setDefaultClassLoader` is required to be called.

The issue that I unfortunately didn't notice until these tests failed is that the tests specify that you can `setDefaultClassLoader` after serializing an object (I'm not sure this functionality is actually used), causing an "incorrect" Kryo instance to be in the pool. Unfortunately the pool doesn't expose a way to clear itself out, thus the var, which clearly doesn't work.

I will work on a solution